storage/innobase/include/ut0counter.h

```diff
@@ -73,6 +73,8 @@ template <typename Type, int N = IB_N_SLOTS>
 struct ib_counter_t {
 	/** Increment the counter by 1. */
 	void inc() { add(1); }
+	ib_counter_t& operator++() { inc(); return *this; }
+	ib_counter_t& operator++(int) { inc(); return *this; }
```

Comment:
This is a postfix increment. You need to store an old value, increment and return the old value.

Reply:
right. in fact, got rid of the postfix increment as the usage of it is limited and the pre-value is not consumed. So no point in invoking a constructor and returning a value that is not being used. Better switch to pre-fix increment operator.